app/src/test/java/mozilla/lockbox/store/RouteStoreTest.kt

```diff
@@ -43,9 +84,37 @@ class RouteStoreTest {
 
     @Test
     fun `dispatched non-routeactions`() {
+        setUpSubject()
         val action = LifecycleAction.UserReset
         dispatcher.dispatch(action)
 
         routeObserver.assertEmpty()
     }
+
+    @Test
+    fun `datastore failed to create backend`() {
+        setUpErrorSubject()
+
+        dispatcher.dispatch(DataStoreAction.Sync)
+        dispatcher.dispatch(OnboardingStatusAction(false))
+
+        routeObserver.assertEmpty()
+    }
+
+    @Test
+    fun `dispatched on datastore`() {
+        setUpSubject()
+        val routesIterator = subject.routes.blockingIterable().iterator()
+
+        dispatcher.dispatch(OnboardingStatusAction(false))
+
+        dispatcher.dispatch(DataStoreAction.Reset)
```

Comment:
Similarly, I'd fake out the `DataStore.State` observable here rather than having the `DataStore` actually do work in this PR.

Reply:
Done!